Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -153,6 +158,31 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian_2nd(
                         Hz_eff += MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0, 0, 1), amrex::IntVect(1, 0, 0), Hz_old);
                     }
 
+                    int mag_exchange = 0;
+                    if (mag_exchange == 1){
+                        // H_exchange
+                        amrex::Real M_exchange_constant = 0.001;
+                        Hx_eff += 2.0 * M_exchange_constant / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0);
+                        Hy_eff += 2.0 * M_exchange_constant / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1);
+                        Hz_eff += 2.0 * M_exchange_constant / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2);
+                    }                    
+
+
+                    int mag_anisotropy = 0;
+                    if (mag_anisotropy == 1){
+                        // H_anisotropy
+                        amrex::Real M_anistropy_constant = 0.001;
+                        amrex::Vector<int> anisotropy_axis = amrex::Vector<int>(3,0);
+                        amrex::Real M_dot_anisotropy_axis = 0.0;
+                        for (int comp=0; comp<3; ++comp) {
+                            M_dot_anisotropy_axis += M_xface(i, j, k, comp) * anisotropy_axis[comp];
+                        }
+                        Hx_eff += - 2.0 * M_anistropy_constant / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx * M_dot_anisotropy_axis * anisotropy_axis[0];
+                        Hy_eff += - 2.0 * M_anistropy_constant / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx * M_dot_anisotropy_axis * anisotropy_axis[1];
+                        Hz_eff += - 2.0 * M_anistropy_constant / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx * M_dot_anisotropy_axis * anisotropy_axis[2];
+                    }                     
```

Comment:
Refer to Page 84 - 91 in Stancil book for anisotropy. 
Anisotropy will possibly give more headache because the coordinate system of the material anisotropy axis maybe different from the external applied field.  
Also note this equation in Aziz's paper only applied to uniaxial anisotropy. We might need to add one additional flag `material_crystalline_type` to select which type of crystalline anisotropy we are looking at, e.g. uniaxial, cubic.